src/components/ec-checkbox/ec-checkbox.vue

```diff
@@ -0,0 +1,209 @@
+<template>
+  <div class="ec-checkbox">
+    <input
+      :id="id"
+      ref="checkboxInput"
+      v-model="inputModel"
+      :aria-describedby="errorId"
+      type="checkbox"
+      class="ec-checkbox__input"
+      :disabled="disabled"
+      v-bind="$attrs"
+      @focus="inputIsFocused = true"
+      @blur="inputIsFocused = false"
+      v-on="$listeners"
+    >
+
+    <div class="ec-checkbox__label-checkbox-wrapper ">
+      <span
+        class="ec-checkbox__check-icon-wrapper"
+        :class="{
+          'ec-checkbox__check-icon-wrapper--focused': inputIsFocused,
+          'ec-checkbox__check-icon-wrapper--checked': inputModel,
+          'ec-checkbox__check-icon-wrapper--checked-and-focused': inputModel && inputIsFocused,
+          'ec-checkbox__check-icon-wrapper--error': (isInvalid || $slots['error-message']) && !inputModel,
```

Comment:
You don't need to check for `$slots['error-message']` because that's part of the `isInvalid`
```
    isInvalid() {
      return !!this.errorMessage || !!this.$slots['error-message'];
    },
```